### PR TITLE
Cleaner build logs

### PR DESCRIPTION
**Pull request by KevinMind**
Fixes: <https://github.com/mozilla/addons/ISSUENUM>

### Description

- Set default shell to bash for docker build with better error handling and logs
- modify some mounts, remove others, simplify the build instructions to reduce error space
- remove single quotes in comments, shouldn't matter but just in case

### Testing

Run a local build, modify one of the `<<EOF` run commands to invalidate the cache. Re-run the build. everything builds okay

### Checklist

- [ ] Add `#ISSUENUM` at the top of your PR to an existing open issue in the mozilla/addons repository.
- [X] Successfully verified the change locally.
- [X] The change is covered by automated tests, or otherwise indicated why doing so is unnecessary/impossible.
- [ ] Add or update relevant [docs](../docs/) reflecting the changes made.

### CODE REVIEW

**Review by eviljeff — APPROVED** [review]
wfm

File: Dockerfile [hunk 1]
```diff
@@ -162,11 +164,17 @@ COPY --from=locales --chown=olympia:olympia ${HOME}/locale ${HOME}/locale
...
+# Add build.py build UUID
+RUN ${HOME}/scripts/generate_build.py > build.py
+
 # version.json is overwritten by CircleCI (see circle.yml).
 # The pipeline v2 standard requires the existence of /app/version.json
 # inside the docker image, thus it's copied there.
 COPY version.json /app/version.json
 
+# Set shell back to sh until we can prove we can use bash at runtime
+SHELL ["/bin/sh", "-c"]
```
Comment (KevinMind) [inline Dockerfile:176]: @diox is there a good reason not to keep bash as the default shell for the image?

Comment (diox) [reply Dockerfile:176]: I imagine it's fine to have `bash` as default.

Comment (KevinMind) [reply Dockerfile:176]: Okay. I will open a separate PR for that so we can easier split it if something does go wrong.

**Pull request merged**